internal/generator/templates/decode.go.tmpl

```diff
@@ -36,12 +36,12 @@
     } else if present {
 
     {{- if $field.Array }}
-        var value ztype.Array[{{if $native.IsMarshaler }}*{{ end }}{{ goType $scope $native.Type -}}, {{ template "array_traits_type.go.tmpl" dict "pkg" $scope "type" $native.Type }}]
+        var value []{{ goType $scope $field.Type }}
```

Comment:
During decoding, we only sometimes know the size of the array (in case it is a fixed array). For non-fixed arrays, the size is calculated during decoding (the first few bits contain the array size).